biapol_utilities/label/_label_overlap_matrix.py

```diff
@@ -17,21 +19,17 @@ def label_overlap_matrix(label_image_x, label_image_y):
     Returns
     -------
     overlap: ND-array, int
-        matrix of pixel overlaps of size [x.max()+1, y.max()+1]
+        matrix of pixel overlaps of size [max(x, y) + 1, max(x, y) + 1], with 
+        entry result[i, j] corresponding to the overlap of label i in image x 
+        with label j in image y
     """
-    # put label arrays into standard form then flatten them 
-#     x = (utils.format_labels(x)).ravel()
-#     y = (utils.format_labels(y)).ravel()
+    
+    
+    # Flatten input arrays
     label_image_x = label_image_x.ravel()
     label_image_y = label_image_y.ravel()
     
-    # preallocate a 'contact map' matrix
-    overlap = np.zeros((1 + label_image_x.max(), 1 + label_image_y.max()), dtype=np.uint)
+    # Calculate confusion matrix
+    overlap = metrics.confusion_matrix(label_image_x, label_image_y)
```

Comment:
Is the function `label_overlap_matrix(a, b)` really necessary, if we also could just call `sklearn.metrics.confusion_matrix(a.ravel(), b.ravel())`? See also my comment in https://github.com/BiAPoL/biapol-utilities/issues/3#issuecomment-962593032.

I explicitly suggest to remove `label_overlap_matrix`. Let me know what speaks against that.

Reply:
Hm...
+ Makes code cleaner and shorter
- If we eventually decide to replace this function by something faster/more appropriate for superlarge images, i.e. GPU-acceleration, we would have to re-introduce it.

Reply:
Then let's remove it for now and potentially later introduce it in case we want to GPU-accelerate something.